dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java

```diff
@@ -321,6 +321,30 @@ public Tool updateContainer(@ApiParam(hidden = true) @Auth User user,
 
     }
 
+    @PUT
+    @Timed
+    @UnitOfWork
+    @Path("/{toolId}/defaultVersion")
+    @ApiOperation(value = "Update the default version of the given tool.", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = Tool.class)
+    public Tool updateDefaultVersion(@ApiParam(hidden = true) @Auth User user,
+                                @ApiParam(value = "Tool to modify.", required = true) @PathParam("toolId") Long toolId,
+                                @ApiParam(value = "Tag name to set as default.", required = true) String tag) {
+        Tool t = toolDAO.findById(toolId);
+        checkEntry(t);
+        checkUser(user, t);
+
+        if (tag != null) {
+            if (!t.checkAndSetDefaultVersion(tag)) {
+                throw new CustomWebApplicationException("Tool version does not exist.", HttpStatus.SC_BAD_REQUEST);
```

Comment:
Nitpick: this is probably more kosher as a 404 than a 400. There's no syntax problem here, its that the version is not found